editor/filesystem_dock.cpp

```diff
@@ -1563,10 +1563,32 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
 }
 
 void FileSystemDock::_update_resource_paths_after_move(const HashMap<String, String> &p_renames, const HashMap<String, ResourceUID::ID> &p_uids) const {
-	// Update the paths in ResourceUID, so that UIDs remain valid.
-	for (const KeyValue<String, ResourceUID::ID> &pair : p_uids) {
-		if (p_renames.has(pair.key)) {
-			ResourceUID::get_singleton()->set_id(pair.value, p_renames[pair.key]);
+	for (const KeyValue<String, String> &pair : p_renames) {
+		// Update UID path.
+		if (p_uids.has(pair.key)) {
+			ResourceUID::get_singleton()->set_id(p_uids[pair.key], pair.value);
+		}
+
+		// Update paths for global classes.
+		int index = -1;
+		EditorFileSystemDirectory *efd = EditorFileSystem::get_singleton()->find_file(pair.key, &index);
+
+		if (!efd->get_file_script_class_name(index).is_empty()) {
+			String lang;
+			for (int i = 0; i < ScriptServer::get_language_count(); i++) {
+				if (ScriptServer::get_language(i)->handles_global_class_type(efd->get_file_type(index))) {
```

Comment:
You can do `const String &type = efd->get_file_type(index)` before the loop.

Reply:
I'm struggling to understand what gets stored there, it returns `"GDScript"` for `.gd` files. Can it be trusted for whether the file is a script of any language?

Reply:
This method returns the file type as recognized by ResourceLoader, so it's reliable.
btw you can use this combined with ClassDB to filter out non-script files, as discussed above.